src/details/ArborX_MinimumSpanningTree.hpp

```diff
@@ -90,6 +95,12 @@ struct FindComponentNearestNeighbors
 
   KOKKOS_FUNCTION void operator()(int i) const
   {
+    auto const n = _bvh.size();
+    auto const component = _labels(i);
+    auto &radius = _radii(component - n + 1);
+    if (radius < _lower_bounds(i - n + 1))
+      return;
```

Comment:
What happened to the version with the scan you showed me?  I thought you said it was faster.

Reply:
I thought so too. But I repeated experiments on Summit yesterday, and could not reproduce. Maybe I ran so many experiments that different results got confused in my head. So, the current patch does not actually seem to help on GPU. On CPU it's better than what I showed you for two reasons: a) we don't do parallel_scan, and b) more importantly, radius may have already been updated by another thread that converged, so it even more threads drop out.